File: typecast-java/src/main/java/io/typecast/exceptions/PaymentRequiredException.java
package io.typecast.exceptions;

/**
 * Exception thrown when the API returns a 402 Payment Required error.
 * 
 * <p>This typically indicates insufficient credits in the account.</p>
 */
public class PaymentRequiredException extends TypecastException {
    
    /**
     * Creates a new PaymentRequiredException.
     * 
     * @param message      the error message
     * @param responseBody the raw response body
     */
    public PaymentRequiredException(String message, String responseBody) {
        super(message, 402, responseBody);
    }

    /**
     * Creates a new PaymentRequiredException with a cause.
     * 
     * @param message      the error message
     * @param responseBody the raw response body
     * @param cause        the underlying cause
     */
    public PaymentRequiredException(String message, String responseBody, Throwable cause) {
        super(message, 402, responseBody, cause);
    }
}

File: typecast-php/src/Exceptions/PaymentRequiredException.php
<?php

declare(strict_types=1);

namespace Neosapience\Typecast\Exceptions;

class PaymentRequiredException extends TypecastException
{
    public function __construct(string $message = 'Payment required', ?\Throwable $previous = null)
    {
        parent::__construct($message, 402, $previous);
    }
}
